src/features/Posts/pages/TagPage/TagPage.tsx

```diff
@@ -0,0 +1,138 @@
+import { useUpdateEffect } from "@react-hookz/web";
+import { useState } from "react";
+import { useFeedQuery, useTagFeedQuery } from "src/graphql";
+import { useInfiniteQuery, usePreload } from "src/utils/hooks";
+import PostsList from "../../Components/PostsList/PostsList";
+import TrendingCard from "../../Components/TrendingCard/TrendingCard";
+import styles from "./styles.module.scss";
+import { Helmet } from "react-helmet";
+import Button from "src/Components/Button/Button";
+import { FiArrowLeft, FiArrowRight } from "react-icons/fi";
+import { capitalize, formatHashtag } from "src/utils/helperFunctions";
+import { createRoute, PAGES_ROUTES } from "src/utils/routing";
+import { Link, useLoaderData } from "react-router-dom";
+import { useAppDispatch } from "src/utils/hooks";
+import { stageStory } from "src/redux/features/staging.slice";
+import { LoaderData } from "./tagPage.loader";
+import IconButton from "src/Components/IconButton/IconButton";
+
+export default function TagPage() {
+  const loaderData = useLoaderData() as LoaderData;
+
+  const tagInfo = loaderData.getTagInfo;
+  const dispatch = useAppDispatch();
+
+  const feedQuery = useTagFeedQuery({
+    variables: {
+      take: 10,
+      skip: 0,
+      sortBy: null,
+      tag: loaderData.getTagInfo.id,
+    },
+  });
+  const { fetchMore, isFetchingMore } = useInfiniteQuery(feedQuery, "getFeed");
+
+  usePreload("PostPage");
+
+  const clickWriteStory = () => {
+    dispatch(
+      stageStory({
+        tags: [
+          {
+            id: tagInfo.id,
+            title: tagInfo.title,
+          },
+        ],
+      })
+    );
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>{`${loaderData.getTagInfo.title} Stories`}</title>
+        <meta
+          property="og:title"
+          content={`${loaderData.getTagInfo.title} Stories`}
+        />
+      </Helmet>
+      <div className={`page-container`}>
+        <div className={`w-full ${styles.grid}`}>
+          <div id="content" className="">
+            <PostsList
+              isLoading={feedQuery.loading}
+              items={feedQuery.data?.getFeed}
+              isFetching={isFetchingMore}
+              onReachedBottom={fetchMore}
+            />
+          </div>
+          <aside id="categories" className="no-scrollbar">
+            <div className="md:overflow-y-scroll sticky-side-element flex flex-col gap-16 md:gap-24">
+              <Link
+                to={PAGES_ROUTES.blog.feed}
+                className="self-start text-body2 text-gray-600"
+              >
+                <IconButton>
+                  <FiArrowLeft />
+                </IconButton>
+              </Link>
+              <h1 className="text-h2 font-bolder">
```

Comment:
lets make this text-body2 like the other headings on the sidebars cause tag names can get pretty long.